lib/teiserver/lobby/libs/lobby_restrictions.ex

```diff
@@ -0,0 +1,280 @@
+
+    cond do
+      # Default ratings
+      consul_state.maximum_rating_to_play >= @rating_upper_bound &&
+          consul_state.minimum_rating_to_play <= 0 ->
+        nil
+
+      # Just a max rating
+      consul_state.maximum_rating_to_play < @rating_upper_bound &&
+          consul_state.minimum_rating_to_play <= 0 ->
+        "Max rating: #{consul_state.maximum_rating_to_play}"
+
+      # Just a min rating
+      consul_state.maximum_rating_to_play >= @rating_upper_bound &&
+          consul_state.minimum_rating_to_play > 0 ->
+        "Min rating: #{consul_state.minimum_rating_to_play}"
+
+      # Rating range
+      consul_state.maximum_rating_to_play < @rating_upper_bound ||
+          consul_state.minimum_rating_to_play > 0 ->
+        "Rating between: #{consul_state.minimum_rating_to_play} - #{consul_state.maximum_rating_to_play}"
+
+      true ->
+        nil
+    end
+  end
+
+  def get_failed_rank_check_text(player_rank, consul_state) do
+    bounds = get_rank_bounds_for_title(consul_state)
+
+    [
+      @splitter,
+      "You don't meet the chevron requirements for this lobby (#{bounds}). Your chevron level is #{player_rank + 1}. Learn more about chevrons here:",
+      "https://www.beyondallreason.info/guide/rating-and-lobby-balance#rank-icons"
+    ]
+  end
+
+  def get_failed_rating_check_text(player_rating, consul_state, rating_type) do
+    bounds = get_rating_bounds_for_title(consul_state)
+    player_rating_text = player_rating |> Decimal.from_float() |> Decimal.round(2)
+
+    [
+      @splitter,
+      "You don't meet the rating requirements for this lobby (#{bounds}). Your #{rating_type} match rating is #{player_rating_text}. Learn more about rating here:",
+      "https://www.beyondallreason.info/guide/rating-and-lobby-balance#openskill"
+    ]
+  end
+
+  @doc """
+  Returns {:ok, nil} or {:error,msg}
+  """
+  def check_rank_to_play(user, consul_state) do
+    state = consul_state
+
+    # Contributors auto pass since their ranks are not defined on playtime. To be fixed seperately.
+    is_contributor? = CacheUser.is_contributor?(user)
+
+    if is_contributor? do
+      {:ok, nil}
+    else
+      cond do
+        state.minimum_rank_to_play != nil and user.rank < state.minimum_rank_to_play ->
+          # Send message
+          msg = get_failed_rank_check_text(user.rank, state)
+          {:error, msg}
+
+        state.maximum_rank_to_play != nil and user.rank > state.maximum_rank_to_play ->
+          # Send message
+          msg = get_failed_rank_check_text(user.rank, state)
+          {:error, msg}
+
+        true ->
+          {:ok, nil}
+      end
+    end
+  end
+
+  @doc """
+  Returns {:ok, nil} or {:error,msg}
+  """
+  def check_rating_to_play(user_id, consul_state) do
+    state = consul_state
+    team_size = state.host_teamsize
+    team_count = state.host_teamcount
+
+    rating_type =
+      cond do
+        team_count > 2 && team_size == 1 -> "FFA"
```

Comment:
Needs to change when Lexon does big/small team split

Reply:
I would leave that as a comment in the code itself.
You could also leave that in the commit message, but for this, you would need a much smaller commit doing pretty much only this action.